images/base/Dockerfile

```diff
@@ -117,64 +120,69 @@ RUN git clone --filter=tree:0 "${CONTAINERD_CLONE_URL}" /containerd \
     && cd /containerd \
     && git checkout "${CONTAINERD_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
     && make bin/ctr bin/containerd bin/containerd-shim-runc-v2 \
-    && go-licenses save --save_path=/_LICENSES \
+    && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES \
         ./cmd/ctr ./cmd/containerd ./cmd/containerd-shim-runc-v2
 
 # stage for building runc
 FROM go-build as build-runc
-ARG GO_VERSION
+ARG TARGETARCH GO_VERSION
 ARG RUNC_VERSION="v1.1.7"
 ARG RUNC_CLONE_URL="https://github.com/opencontainers/runc"
 RUN git clone --filter=tree:0 "${RUNC_CLONE_URL}" /runc \
     && cd /runc \
     && git checkout "${RUNC_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
     && make runc \
-    && go-licenses save --save_path=/_LICENSES .
+    && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES .
 
 # stage for building crictl
 FROM go-build as build-crictl
-ARG GO_VERSION
+ARG TARGETARCH GO_VERSION
 ARG CRI_TOOLS_CLONE_URL="https://github.com/kubernetes-sigs/cri-tools"
 ARG CRICTL_VERSION="v1.26.1"
 RUN git clone --filter=tree:0 "${CRI_TOOLS_CLONE_URL}" /cri-tools \
     && cd /cri-tools \
     && git checkout "${CRICTL_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
     && make BUILD_BIN_PATH=./build crictl \
-    && go-licenses save --save_path=/_LICENSES ./cmd/crictl
+    && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES ./cmd/crictl
 
 # stage for building cni-plugins
 FROM go-build as build-cni
-ARG GO_VERSION
+ARG TARGETARCH GO_VERSION
 ARG CNI_PLUGINS_VERSION="v1.2.0"
 ARG CNI_PLUGINS_CLONE_URL="https://github.com/containernetworking/plugins"
 RUN git clone --filter=tree:0 "${CNI_PLUGINS_CLONE_URL}" /cni-plugins \
     && cd /cni-plugins \
     && git checkout "${CNI_PLUGINS_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
     && mkdir ./bin \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
```

Comment:
I'm not sure if we actually want CGO for these.

We do for containerd, runc. Maybe not for the others.
Thought most of them can already control that in their makefile etc. This is the only set we're just `go build` ing.

Reply:
I don't think it will cause problems, anyhow, and it looks like the usual builds do not disable CGO.

CGO is disabled by default when attempting to cross compile with go by setting GOARCH though

https://github.com/containernetworking/plugins/blob/38f18d26ecfef550b8bac02656cc11103fd7cff1/build_linux.sh#L20

Reply:
AFAIK this will impact only the libc binding  and mostly dns resolution, since we control the image it should be relatively safe, I honestly can't say what is better, but the cni plugins and crictl are small programs

Reply:
crictl we're using the makefile anyhow so if they typically release/test with static that should remain the case.

for this one where we're `go build` ourselves it's worth double checking if that build is comparable to the released binaries. AFAICT it is.

Reply:
> since we control the image it should be relatively safe

yeah, that's a motivating factor in #3201 proceeding this, ran into some issues working on an image build and glibc symbol mismatch with external cgo binaries